src/Raven.Server/Documents/Indexes/MapIndexBase.cs

```diff
@@ -58,7 +58,7 @@ public override int HandleMap(IndexItem indexItem, IEnumerable mapResults, Index
                 mustDelete = _filters.Add(indexItem.LowerId) == false;
             }
 
-            if (mustDelete)
+            if ((LastDatabaseEtagOnIndexCreation < indexItem.Etag || indexItem.Referenced) && mustDelete)
```

Comment:
You could store last etag per each of the collection needed, then you could also apply that for references?

Reply:
I need the etag of the _referenced_ document when indexing a reference.
Actually, we can skip indexing references entirely until we reach the `LastDatabaseEtagOnIndexCreation`